tests/lib.rs

```diff
@@ -40,50 +40,86 @@ fn random_bytes(length: usize) -> Vec<u8> {
 
 const DATA_SIZE : u64 = 20 * 1024 * 1024;
 
+pub struct Entry {
+  name: Vec<u8>,
+  data: Vec<u8>
+}
+
 pub struct MyStorage {
-  temp_dir : TempDir
+  entries: Vec<Entry>
 }
 
 impl MyStorage {
   pub fn new() -> MyStorage {
-    MyStorage { temp_dir: match TempDir::new("encrypt_storage") {
-        Ok(dir) => dir,
-        Err(why) => panic!("couldn't create temporary directory: {}", why)
-    } }
+    MyStorage { entries: Vec::new() }
+  }
+
+  pub fn has_chunk(&self, name: Vec<u8>) -> bool {
+    for entry in self.entries.iter() {
+      if entry.name == name { return true }
+    }
+    false
   }
 }
 
 impl Storage for MyStorage {
   fn get(&self, name: Vec<u8>) -> Vec<u8> {
-    let file_name = String::from_utf8(name).unwrap();
-    let file_path = self.temp_dir.path().join(Path::new(&file_name)); 
-    let mut f = match File::open(&file_path) {
-        // The `desc` field of `IoError` is a string that describes the error
-        Err(why) => panic!("on get couldn't open: {}", why),
-        Ok(file) => file,
-    };
-    let mut s = String::new();
-    //f.read_to_string(&mut s); put f into a string
-    match f.read_to_string(&mut s){
-        Err(why) => panic!("on get couldn't read: {}", why),
-        Ok(_) => print!("contains:\n{}", s),
+    for entry in self.entries.iter() {
+      if entry.name == name { return entry.data.to_vec() }
     }
-    s.into_bytes()
+
+    vec![]
   }
 
-  #[allow(unused_must_use)]
   fn put(&mut self, name: Vec<u8>, data: Vec<u8>) {
-    let file_name = String::from_utf8(name).unwrap();
-    let file_path = self.temp_dir.path().join(Path::new(&file_name)); 
-    let mut f = match File::create(&file_path) {
-        // The `desc` field of `IoError` is a string that describes the error
-        Err(why) => panic!("on put couldn't open: {}", why),
-        Ok(file) => file,
-    };
-    f.write_all(&data);
+    self.entries.push(Entry { name : name, data : data })
   }
 }
 
+// pub struct MyStorage {
```

Comment:
Temporarily swapped disk based Storage with memory based Storage, to avoid the bug that shows up with this disk based Storage